### PR TITLE
Add support for reducing the UPM preview version length

### PR DESCRIPTION
**Pull request by davidkline-ms**
This change reduces the length of the UPM preview version by trimming the year from the build number.

Before:  4.3.2-preview.20200925.1
After: 4.3.2-preview.0925.1

### CODE REVIEW

File: pipelines/templates/tasks/upmpackages.yml [hunk 1]
```diff
@@ -6,11 +6,18 @@ parameters:
...
+  value: ""
+
 steps:
 - task: NodeTool@0
   inputs:
     versionSpec: '12.18.0'
-    
+
+- powershell: |
+  $previewVersion = ${{ parameters.buildNumber }}.Substring(4)
```
Comment (keveleigh) [inline pipelines/templates/tasks/upmpackages.yml:19]: I think this needs to be explicitly stored in the yml variable:

```suggestion
    $previewVersion = ${{ parameters.buildNumber }}.Substring(4)
    echo "##vso[task.setvariable variable=previewVersion"]$previewVersion
```

File: pipelines/templates/tasks/upmpackages.yml [hunk 2]
```diff
@@ -21,7 +28,7 @@ steps:
         -ProjectRoot ${{ parameters.projectRoot }}
         -OutputDirectory ${{ parameters.outputDirectory }}
         -Version ${{ parameters.version }}
-        -BuildNumber ${{ parameters.buildNumber }}
+        -BuildNumber $(parameters.previewVersion)
```
Comment (keveleigh) [inline pipelines/templates/tasks/upmpackages.yml:31]: I don't think variables need the `parameters.` part

```suggestion
        -BuildNumber $(previewVersion)
```

File: pipelines/templates/tasks/upmpackages.yml [hunk 3]
```diff
@@ -33,7 +40,7 @@ steps:
...
```
Comment (keveleigh) [inline pipelines/templates/tasks/upmpackages.yml:43]:
```suggestion
        -BuildNumber $(previewVersion)
```

**Comment by wiwei** [comment]
Is there some background for why we don't use the azure pipelines build number config itself:

https://docs.microsoft.com/en-us/azure/devops/pipelines/process/run-number?view=azure-devops&tabs=yaml

i.e. this change as-is I think does some string parsing off the generated output of that build number - why not just update the specific build pipeline to use the correct value throughout the pipeline?

**Comment by davidkline-ms** [comment]
> i.e. this change as-is I think does some string parsing off the generated output of that build number - why not just update the specific build pipeline to use the correct value throughout the pipeline?

Good question. Will look into that

**Comment by davidkline-ms** [comment]
determined to not make this change. this impacts preview releases only

**Pull request closed without merging**